packages/toolkit/src/view/model/ModelsList.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+
+import {
+  InfiniteData,
+  QueryObserverResult,
+  RefetchOptions,
+} from "@tanstack/react-query";
+import { CardModel } from "../../components/card-model/CardModel";
+import { ListUserModelsResponse, Model, Nullable } from "../../lib";
+import { CardModelSkeleton } from "../../components/card-model/Skeleton";
+
+export type ModelsListProps = {
+  models: Model[];
+  accessToken: Nullable<string>;
+  onModelDelete: (
+    options?: RefetchOptions | undefined
+  ) => Promise<
+    QueryObserverResult<InfiniteData<ListUserModelsResponse, unknown>, Error>
+  >;
+  isLoading: boolean;
+};
+
+export const ModelsList = (props: ModelsListProps) => {
+  const { models, accessToken, onModelDelete, isLoading } = props;
```

Comment:
@orangecoloured We have accessToken and enbaedQuery in our global store. 

You can access them with useInstillStore, please look for pipeline related component for how to use it